src/main.rs

```diff
@@ -495,6 +497,18 @@ extern "C" fn kernel_init(hart_id: u64, fdt_addr: u64) {
         }
     };
 
+    // Initialize global Umode state.
+    UmodeTask::init(umode_elf);
+    // Setup U-mode task for this CPU.
+    UmodeTask::setup_this_cpu();
+
+    // Simple test: run U-mode until the first `ecall`.
+    let res = UmodeTask::get()
+        .activate()
+        .expect("Could not activate U-mode")
+        .run();
+    println!("umode result: {:?}", res);
```

Comment:
left over debug print?

Reply:
Hm. This is just temporary code to check that things work. The unwrap was reviewed as a bit extreme in previous iteration (not by your review). So at least I'd have a print.

Reply:
Switched back to `unwrap()`.